### ISSUE TITLE
pagemodel.py template_choices and DEFAULT_TEMPLATE aren't being used

### ISSUE THREAD

**Issue by ZippoLag** [django-cms/django-cms#6520]
### Summary
I was trying to override `Page`'s `template_choices` and `DEFAULT_TEMPLATE` so I can limit the options according to a page's parent, but it appears that in at least 3 key places where those attributes should be used, `get_cms_setting('TEMPLATES')` is used instead, and so the selectable templates displayed are never limited like I intended.

The places I've found where this happens are the `get_template` and `get_template_name` methods. Also in `cms_toolbars.py` for the `# templates menu` section `get_cms_setting('TEMPLATES')` is used rather than `self.page.template_choices`.

More details about what I was trying to pull of can be found, here: https://stackoverflow.com/questions/52335541/how-to-limit-django-cms-template-choices-based-on-page-parent
### Expected behaviour
`template_choices` and `DEFAULT_TEMPLATE` should define which templates are available for a Page in the CMS.

### Actual behaviour
`get_cms_setting('TEMPLATES')` is always used instead.

### Environment
* Python version: 2.7.x
* Django version: 1.11.x
* django CMS version: 3.4.6

**Comment by othane** [comment]
Any progress on this? I would also like to define the default and available templates for a page based on the parent.

**Comment by ZippoLag** [comment]
Nope, thanksully I could get the client to accept a "I'm sorry but there's no way for me to do this unless you want to pay for the time it'd take me to dive into the source of django-cms" so they just live without the feature. I'd love to have the free time to do it though.

El mié., 25 de sep. de 2019 a la(s) 19:50, Oliver Thane (notifications@github.com<mailto:notifications@github.com>) escribió:

Any progress on this? I would also like to define the default and available templates for a page based on the parent.

**Comment by victor-yunenko** [comment]
Hi @ZippoLag,

Thank you for your interest in django CMS and thank you especially for raising this issue!

The django CMS project is entirely community-driven. This means that we rely on the support of contributors. Would you like to help us solve this issue? We would be very happy to receive a Pull Request from you.

Here are the next steps:
- Have a look at our contributor guidelines
- Join #contributor on Slack
- Our Community Manager will get in touch with you

If you have any questions please feel free to post a comment here.

All the best and hope to hear from you.
